File: tests/openai-stream.py
import openai
openai.api_key = "dummy"
openai.api_base = "https://niansuhai-bingo.hf.space" # You can change this to your own deployed service. The bingo service version needs to be >= 0.9.0

# create a chat completion
completion = openai.ChatCompletion.create(model="Creative", stream=True, messages=[{"role": "user", "content": "Hello"}])
for chat_completion in completion:
    # print the completion
    print(chat_completion.choices[0].message.content, end="", flush=True)

File: tests/openai-stream.ts
import OpenAI from 'openai';

const openai = new OpenAI({
  apiKey: 'dummy',
  baseURL: 'http://127.0.0.1:3000' // Change this to the service address you deployed yourself
});

async function start() {
  const completion = await openai.chat.completions.create({
    messages: [
      { role: 'user', content: 'Hello' },
    ],
    model: 'bing',
    stream: true,
  });
  for await (const part of completion) {
    process.stdout.write(part.choices[0]?.delta?.content || '');
  }
}

start()
